### DISCUSSION TITLE
[Bitcoin] Runes special flags incompatibility with Phantom

### DISCUSSION THREAD

**Discussion by 0xboga** [phantom/docs#265] [Help & Support]
### General observations:

1. All actions with error on Phantom provider have successfully been completed on:

    -  XVERSE
    -  MagicEden Wallet
    -  Unisat
    -  Leather ( ex Hiro )

2. All action with error on Phantom provider behaved the same across different websites

    - Tested with Magisat and MagicEden PSBTs.

### Issues

Transactions that involves inputs / outputs with an `OP_RETURN` code specific for runes management do not properly work:

**[i]** Simulation fails when signing split runes transaction

**[ii]** Signing a partial transaction with an input coming from a successful split rune transaction ( i.e contains `OP_RETURN` specific for runes ) throws error

**[iii]** ( Probably related to [i] and [ii] ) Completing a partial transaction signed by another wallet throws error

### Example

https://magisat.io/isolate/runes

### Steps to Reproduce

[i] **Steps for simulation fail:** 
1. Retrieve a PSBT (splitting runes from one input into two outputs)
`cHNidP8BAP00AQIAAAADnWzh23jMlGdISZTQbo7uJkBjGX2GkuB/DHrQbvVZF0wBAAAAAP////9aIMNPLRatXbwjknppMKjFCi03vlehRHwQY7clcZyuCwIAAAAA/////9GB2u6yciwkkB8mKcPG1Wm4elMK+fQkiVaxn05D10Q8AwAAAAD/////BAAAAAAAAAAAMWpdLgCM1TObEdW+58ip1arVqtWq1arVqtWqVQEAANa+58ip1arVqtWq1arVqtWqVQIiAgAAAAAAACJRIB5A3RnUcAbKKjGaNl1gvKxsTViVBWFsLB8WtzwDHm6gIgIAAAAAAAAiUSAeQN0Z1HAGyioxmjZdYLysbE1YlQVhbCwfFrc8Ax5uoN9wAAAAAAAAFgAUMEWo+PaOLEE7MuX5wQtPrM6eoBoAAAAAAAEA/ZYBAgAAAAABAppgd2JUQAGElugqFzHpmnDG0s84sCIP3sk0vzdWUH4gAAAAAAD/////jAd8hNVxtnhqh2H/3oc2IA5vGeHp8jDEU7vvsdpTZPIDAAAAAP////8EAAAAAAAAAAAzal0wAIzVM5sRq/3OkdOq1arVqtWq1arVqtWqAQEAAKv9zpHTqtWq1arVqtWq1arVqgECIgIAAAAAAAAiUSAeQN0Z1HAGyioxmjZdYLysbE1YlQVhbCwfFrc8Ax5uoCICAAAAAAAAIlEgTAaSB4ltHZvCEVp0PmR3vsSgUVnki5m1sOIRJiyr3+3/AgAAAAAAABepFKEFNpkjagVcMANXIIgmD2nJ1vCehwFBuQ/NIH7G0OxqPEX7VmCmfyqNXF2vHXvKYXMEiGpJlkdWxgAecCS6yfQWcs5fb5FnS9oTkebt05s0SOIFKFgjRAEBQSTTbQkaOrHR4TUszPIHklR3Pc66Zb/W7my6EOmPnO4x+fXeLkvcyAZYsft2GUxJ15B5PSZCewiJrWREUEpdCdwBAAAAAAEBKyICAAAAAAAAIlEgHkDdGdRwBsoqMZo2XWC8rGxNWJUFYWwsHxa3PAMebqABAwQBAAAAARcgLKFymvz3XxsyZTDIcPvZI+VTCfDJ7FKr9gJbbzJNyu0AAQD92AICAAAAAAEEmmB3YlRAAYSW6CoXMemacMbSzziwIg/eyTS/N1ZQfiABAAAAAP////8siJ/lZ+lH+FQQwEpAhTlFnQfWczcDLyK2GvqXtjXtgQIAAAAA/////1DFCFpkOdRienhGen2Qoqa4tuHynAIiQnsOwni1zsofAQAAAAD/////wBgXM8dMkVgJ9bsTxkWGacpa8z6xYqyRSqlFdKSqN2EAAAAAAP////8GCgYAAAAAAAAiUSCLG/RhedET+TXTMCsGC3UGF46kT8D/Cjlx3j4Pv9QbR+gDAAAAAAAAIlEgixv0YXnRE/k10zArBgt1BheOpE/A/wo5cd4+D7/UG0eAPgAAAAAAABYAFDBFqPj2jixBOzLl+cELT6zOnqAaWAIAAAAAAAAiUSCLG/RhedET+TXTMCsGC3UGF46kT8D/Cjlx3j4Pv9QbR1gCAAAAAAAAIlEgixv0YXnRE/k10zArBgt1BheOpE/A/wo5cd4+D7/UG0e2bAAAAAAAACJRIIsb9GF50RP5NdMwKwYLdQYXjqRPwP8KOXHePg+/1BtHAUBjYWL0koYt6gG5Y27kX0AUVHs7AeLwuZs2aYqnH35BWQ8Wu49po48NbgJW1xMPinbHReBAdcn0JRd+yfM2V70NAUBsdgb2JpJnKKjqmxkbZyMPUWoKiHfY1f2u0cCUx6nseFUt5b1kOfBGe551G47becGnpRM7LiRjv7EKoDN/Rw/aAkgwRQIhAPigoCybUEseUIxjosaxynBugwdxXyXzwtOSTqqyZ8e6AiAwKuSzv6asNOg8N1dGaedX28e2cO8ZYbtUm3pU7P8jSYMhAxPIsldVhuqbS8eq99DrTEzcABBCa51CngPNp03J6dRGAUAhOJvc3EIdlfLAOAk3qbBtOGCX9BtsnRYIfuJz1Nk+ur5CoMuwOK7DOaU82yUq1Ko9XaojkigO2R7biKjrJk7qAAAAAAEBH4A+AAAAAAAAFgAUMEWo+PaOLEE7MuX5wQtPrM6eoBoBAwQBAAAAAAEA/YsBAgAAAAABAvkkcuqXbM0T2xbp7kOyBePEI03vvKj/Rd21+PCzynfJAwAAAAD/////qiGMVpJulSxMjGeseE6KSlYbvFKx2NBcykc2vzZJWQIFAAAAAP////8EIgIAAAAAAAAWABQwRaj49o4sQTsy5fnBC0+szp6gGugDAAAAAAAAF6kUoQU2mSNqBVwwA1cgiCYPacnW8J6HiBMAAAAAAAAXqRTE4ReNbchCbr2c1+5k8xgbagO6wYfUQQAAAAAAABYAFDBFqPj2jixBOzLl+cELT6zOnqAaAkgwRQIhAPufDqS4hsu1JmKnEbjvGPJl6Ns79LkdQQ1uK61mrblLAiBmsHOxAlwsJTJVzJPODHjKrUxHD3DhIHC6Gx/SeIFE0QEhAxPIsldVhuqbS8eq99DrTEzcABBCa51CngPNp03J6dRGAUFVqfq1Ou8+VeWpkPP6XW/Mr7/n05y2Zp9eFNpF3rHyIoMyEghm0hcXWQt2Jygsm+X7GEc/IPE0bV2nTd77L1DVgwAAAAABAR/UQQAAAAAAABYAFDBFqPj2jixBOzLl+cELT6zOnqAaAQMEAQAAAAAAAAAA`

     Associated pub key: `032ca1729afcf75f1b326530c870fbd923e55309f0c9ec52abf6025b6f324dcaed`

2. Call `signPSBT` method on Phantom provider.
3. Transaction simulation breaks
![Screenshot from 2024-06-26 12-15-39](https://github.com/user-attachments/assets/ae708ca0-dfb2-43c7-8b99-a75518dd998c) 
**Obs:** Additionally if the input comes from an `OP_RETURN` output - clicking confirm on the failed simulation transaction will result in an error similar to [ii].

[ii] **Signing a partial transaction with an input coming from a successful split rune transaction ( i.e contains `OP_RETURN` specific for runes ) throws error:** 

1. Retrieve a PSBT for partial transaction ( only one input and one output signed with sighash 131 SINGLE | ANYONECANPAY ):
`cHNidP8BAFICAAAAAZ1s4dt4zJRnSEmU0G6O7iZAYxl9hpLgfwx60G71WRdMAQAAAAD/////AQDh9QUAAAAAFgAUMEWo+PaOLEE7MuX5wQtPrM6eoBoAAAAAAAEA/ZYBAgAAAAABAppgd2JUQAGElugqFzHpmnDG0s84sCIP3sk0vzdWUH4gAAAAAAD/////jAd8hNVxtnhqh2H/3oc2IA5vGeHp8jDEU7vvsdpTZPIDAAAAAP////8EAAAAAAAAAAAzal0wAIzVM5sRq/3OkdOq1arVqtWq1arVqtWqAQEAAKv9zpHTqtWq1arVqtWq1arVqgECIgIAAAAAAAAiUSAeQN0Z1HAGyioxmjZdYLysbE1YlQVhbCwfFrc8Ax5uoCICAAAAAAAAIlEgTAaSB4ltHZvCEVp0PmR3vsSgUVnki5m1sOIRJiyr3+3/AgAAAAAAABepFKEFNpkjagVcMANXIIgmD2nJ1vCehwFBuQ/NIH7G0OxqPEX7VmCmfyqNXF2vHXvKYXMEiGpJlkdWxgAecCS6yfQWcs5fb5FnS9oTkebt05s0SOIFKFgjRAEBQSTTbQkaOrHR4TUszPIHklR3Pc66Zb/W7my6EOmPnO4x+fXeLkvcyAZYsft2GUxJ15B5PSZCewiJrWREUEpdCdwBAAAAAAEBKyICAAAAAAAAIlEgHkDdGdRwBsoqMZo2XWC8rGxNWJUFYWwsHxa3PAMebqABAwSDAAAAARcgLKFymvz3XxsyZTDIcPvZI+VTCfDJ7FKr9gJbbzJNyu0AAA==`

      Associated pub key: `032ca1729afcf75f1b326530c870fbd923e55309f0c9ec52abf6025b6f324dcaed`

2. Call `signPSBT` method on Phantom provider.
3. Wallet pops - click sign
4. Phantom provider fails with error:
![09c14437-119b-48a5-8dec-4017cb5ffaf4](https://github.com/user-attachments/assets/99473108-5c0c-4edb-ab86-ef56049b238c)

[iii] **Completing a partial transaction signed by another wallet throws error:**
1. Retrieve a PSBT that would complete a transaction similar to [ii] ( using same public key that wouldn't work on Phantom provider, but signed on another provider i.e XVERSE )
2. Call `signPSBT` method on Phantom provider.
3. Similar error occurs as in [ii]

### Phantom Version

24.13.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hey @0xboga,

I'm looking into this to determine if this is a bug on our end or something we don't support. Will follow up once I have more info.
